src/schema/sources.ts

```diff
@@ -706,15 +730,21 @@ const addNewSourceMember = async (
 
 export const getPermissionsForMember = (
   member: Pick<SourceMember, 'role'>,
-  source: Pick<SquadSource, 'memberPostingRank'>,
+  source: Pick<SquadSource, 'memberPostingRank' | 'memberInviteRank'>,
 ): SourcePermissions[] => {
-  const permissions =
+  let permissions =
     roleSourcePermissions[member.role] ?? roleSourcePermissions.member;
   const memberRank =
     sourceRoleRank[member.role] ?? sourceRoleRank[SourceMemberRoles.Member];
 
   if (memberRank < source.memberPostingRank) {
-    return permissions.filter((item) => item !== SourcePermissions.Post);
+    permissions = permissions.filter((item) => item !== SourcePermissions.Post);
+  }
+
+  if (memberRank < source.memberInviteRank) {
+    permissions = permissions.filter(
+      (item) => item !== SourcePermissions.Invite,
+    );
   }
```

Comment:
Maybe reduce is better as it will not use a let?
Generally we try to avoid them as they might cause leaks down the road and are not super explicit

Reply:
Would not worry about memory leaks here since value is contained inside this function. My main goal here was to do no iterations if we don't need to and I think with reduce you would at least have to double check conditions to retain this behaviour. I do wonder how you would structure this with `reduce`?

Reply:
Chris gave another idea, implemented here 62955e1fd33862c1ca2812fb2801a4c5359a2da2